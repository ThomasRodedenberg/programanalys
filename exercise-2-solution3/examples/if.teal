fun main () = {
    var x : nonnull int := 4; // N 2 27 =, N 2 5 +
    if (0 < 1) { // N 3 8 =, N 3 12 =, NC 3 7 ++
       x := null; // N 4 12 =, N 4 7 0
    } else {
       x := 1; // N 6 12 =, N 6 7 +
    }
    var y : nonnull int := x;
    return x; // N 0 0 =
}