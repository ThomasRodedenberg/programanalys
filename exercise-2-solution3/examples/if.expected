N 2 4 +
NC 3 7 ++
N 4 7 0
N 6 7 +
N 8 4 ?
